src/prefect/flow_engine.py

```diff
@@ -726,3 +732,14 @@ def run_flow(
         return run_flow_async(**kwargs)
     else:
         return run_flow_sync(**kwargs)
+
+
+def _flow_parameters(
```

Comment:
do we need merge logic that takes combines both the provided param values with the defaults on the function?

Reply:
That's a great point. I'd thought about it and tested it out through the UI and that works, but looking at the CLI version it fails so I think we do in fact need a merge here. I'll take that on.